.github/workflows/ci.yml

```diff
@@ -0,0 +1,38 @@
+name: ci
+
+on:
+  push:
+    branches: # Trigger only on the main branch to avoid duplicate runs on PR branches
+      - master
+  pull_request: # Trigger All PRs
+
+jobs:
+  rspec:
```

Comment:
Would it be helpful to cancel previous github action runs as well? https://github.com/styfle/cancel-workflow-action Or do the specs run pretty quickly so it doesn't matter that much?

Reply:
Ah, thanks! Good idea.

Reply:
Actually, gonna merge for now. We can keep an eye on this if the build gets too slow we can add this integration